app/models/git_repository.rb

```diff
@@ -39,6 +39,20 @@ def commit_from_ref(git_reference)
     description.split('-').last.sub(/^g/, '')
   end
 
+  def tag_from_ref(git_reference)
+    Dir.chdir(repo_cache_dir) do
+      tag = IO.popen(['git', 'describe', '--tags', git_reference], err: [:child, :out]) do |io|
```

Comment:
you can always do
silence_stderr { test }

but either way is fine I guess

On Sat, Mar 21, 2015 at 12:26 AM, Gabe Martin-Dempesy <
notifications@github.com> wrote:

> In app/models/git_repository.rb
> <https://github.com/zendesk/samson/pull/308#discussion_r26884327>:
>
> > @@ -39,6 +39,20 @@ def commit_from_ref(git_reference)
> >      description.split('-').last.sub(/^g/, '')
> >    end
> >
> > +  def tag_from_ref(git_reference)
> > +    Dir.chdir(repo_cache_dir) do
> > +      tag = IO.popen(['git', 'describe', '--tags', git_reference], err: [:child, :out]) do |io|
>
> The test output covering the nil case gets an error dumped to stderr that
> I was trying to avoid.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/308/files#r26884327>.
>